### PR TITLE
Fix lambda completion inside `new()` arguments

### LINKED ISSUE #47662: Annoying completion inside implicit object creation syntax
Typing `new(p => ...` auto-completes to `new(prop => ...`.

Thanks @devhl-labs  for pointing this out.
https://github.com/dotnet/roslyn/issues/42556#issuecomment-691576992

FYI @alrz

### PR DESCRIPTION
**Pull request by alrz**
Fixes #47662

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 1]
```diff
@@ -386,7 +386,7 @@ private IEnumerable<TypeInferenceInfo> InferTypeInConstructorInitializer(Constru
                 return InferTypeInArgument(index, methods, argument, parentInvocationExpressionToTypeInfer: null);
             }
 
-            private IEnumerable<TypeInferenceInfo> InferTypeInObjectCreationExpression(ObjectCreationExpressionSyntax expression, SyntaxToken previousToken)
+            private IEnumerable<TypeInferenceInfo> InferTypeInObjectCreationExpression(BaseObjectCreationExpressionSyntax expression, SyntaxToken previousToken)
```
Comment (alrz) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs:389]: This one isn't hit for this fix but I cover it anyways.

Scanning through the type inferred, there seem to be a couple more ObjCreation references that we might need to update.

**Review by CyrusNajmabadi — APPROVED** [review]
Thanks!

**Comment by alrz** [comment]
> `BuddyAssigned`

It's not. @jinujoseph Can you please assign this? Thanks.

**Comment by CyrusNajmabadi** [comment]
Thank you!

**Pull request merged**